Sources/Baggage/BaggageContext.swift

```diff
@@ -41,11 +41,24 @@ public struct BaggageContext {
         get {
             self._storage[ObjectIdentifier(key)]?.forceUnwrap(key)
         } set {
-            self._storage[ObjectIdentifier(key)] = newValue == nil ? nil : ValueContainer(value: newValue!)
+            self._storage[ObjectIdentifier(key)] = newValue.map {
+                // TODO: consider if keys should allow optional string names; static property on the type?
+                ValueContainer(keyName: String(describing: key.self), value: $0)
```

Comment:
iffy about the `ValueContainer` _having_ the name -- that means that:
- every time we _set_ values we need to materialize the string description of the type -- that may be expensive. Again, let's only do it when we can i.e. we'll need to store the key type...

We can do this if we use a special:

```swift
struct AnyKey: Hashable { 
  let keyType: Any.Type
  var id: ObjectIdentifier { ObjectIdentifier(self.keyType) } // this is cheap
  let name: String 
  init<Key>(_ key: Key) where Key: BaggageContextKey { 
    // we can do the if the Key has a static name we could store it; otherwise we can do the description() call here
    ...
   }

  // (hash and equals by the Objectidentifier _only_)
}
```

as key in our `_storage` this way we can use the Any.Type to get its name when we need to (string describing).
We could also experiment with static properties for the names if we needed to...?

Related notes:

- "Strings under 15 utf8 bytes doesn't heap allocate (on 64 bit)" 
  - good for us, most ids should be pretty short i hope
- `Objectidentifier(type)` is cheap and boils down to a (supported) unsafeBitCast